### DISCUSSION TITLE
How to contribute to this project?

### DISCUSSION THREAD

**Discussion by sandboxws** [fwcd/kotlin-language-server#361] [Q&A]
First of all please allow me to thank everyone who is maintaining or contributing to this project.
Second of all, I'm very much interested in helping with this project but I'm currently stuck and I need help with the following:

* How can I test changes I've made to the language server locally from my neovim editor?
  * Basically, I'm trying to get neovim to connect to the server running manually through `gradlew :server:run`  instead of having to build a binary everytime I make a change.

Thanks,
Ahmed

**Comment by fwcd** [comment]
Hi, that's an interesting question. To run the language server directly via a build command you would have to make sure to suppress all build output (since the language server uses stdio to communicate by default), e.g. by using

```sh
./gradlew -q --warning-mode none --console plain :server:run
```

or by picking a different transport method (e.g. TCP). Also you would need to make sure that this command is executed from the `kotlin-language-server` repo directory and would have to be careful to avoid any compilation errors, since these would most likely confuse the LSP client.

Building the server first (e.g. via `./gradlew :server:installDist`) and then pointing your LSP client (Neovim) to the built executable (at `server/build/install/server/bin/kotlin-language-server`) is what I would recommend instead, but you are of course free to try out whatever works best for you.

**Comment by fwcd** [comment]
By the way, something you may find interesting is Gradle's [continuous builds](https://blog.gradle.org/introducing-continuous-build), which watch the file tree for changes and continuously rebuild your project when anything changes. You could for example launch a continuous build using

```sh
./gradlew -t :server:installDist
```

and then simply reload your LSP client whenever you want to see the changes.
